tracetypes/org.eclipse.tracecompass.incubator.opentracing.core/src/org/eclipse/tracecompass/incubator/opentracing/core/analysis/callstack/AsincCallStackStateProvider.java

```diff
@@ -0,0 +1,209 @@
+            long prevTime = fPrevEvent.firstKey();
+            List<Integer> quarks = Objects.requireNonNull(fPrevEvent.get(fPrevEvent.firstKey()));
+            ss.modifySpanAttribute(prevTime, (Object) null, quarks.get(0), quarks.get(1));
+            fPrevEvent.remove(fPrevEvent.firstKey());
+        }
+
+        String processName = event.getContent().getFieldValue(String.class, IOpenTracingConstants.TRACE_ID);
+
+        int processId = getProcessId(event);
+        if (processName == null) {
+            processName = (processId == UNKNOWN_PID) ? UNKNOWN : Integer.toString(processId);
+        }
+
+        int pq = ss.getQuarkAbsoluteAndAdd(PROCESSES, processName);
+        ss.updateOngoingState(TmfStateValue.newValueInt(processId), 1);
+
+        String Opname = String.valueOf(TmfTraceUtils.resolveAspectOfNameForEvent(event.getTrace(), "Name", event)); //$NON-NLS-1$
+        String spanId = event.getContent().getFieldValue(String.class, IOpenTracingConstants.SPAN_ID);
+        String parentId = event.getContent().getFieldValue(String.class, IOpenTracingConstants.REFERENCES + "/CHILD_OF"); //$NON-NLS-1$
+
+        int callStackQuark = ss.getQuarkRelativeAndAdd(pq, InstrumentedCallStackAnalysis.CALL_STACK);
+        int stackDepth = getStackDepth(parentId);
+
+        Object functionEntryName = functionEntry(spanId, parentId, Opname);
+        stackDepth++;
+
+        int spanQuark = ss.getQuarkRelativeAndAdd(callStackQuark, String.valueOf(stackDepth));
+
+        ss.modifySpanAttribute(timestamp, functionEntryName, fStateQuark, spanQuark);
+
+        fStackDepthMap.put(spanId, stackDepth);
+
+        List<Integer> quarksList = Arrays.asList(fStateQuark, spanQuark);
+        fPrevEvent.put(timestamp + duration, quarksList);
+        fStateQuark++;
+
+    }
+
+    protected @Nullable String getProcessName(ITmfEvent event) {
+
+        Long fieldValue = event.getContent().getFieldValue(Long.class, "trace_id");
+        if (fieldValue == null) {
+            fieldValue = event.getContent().getFieldValue(Long.class, "trace_id_low");
+        }
+
+        return fieldValue == null ? "eduroam" : Long.toHexString(fieldValue);
+    }
+
+    protected int getProcessId(ITmfEvent event) {
+        Long resolve = event.getContent().getFieldValue(Long.class, "trace_id");
+        return resolve == null ? -1 : resolve.intValue();
+    }
+
+    private int getStackDepth(String parentId) {
+        Integer stackDepth = fStackDepthMap.get(parentId);
+        if (stackDepth == null) {
+            stackDepth = 0;
+        }
+        if (stackDepth >= MAX_STACK_DEPTH) {
+            /*
+             * Limit stackDepth to 100000, to avoid having Attribute Trees grow
+             * out of control due to buggy insertions
+             */
+            String message = " Stack limit reached, not pushing"; //$NON-NLS-1$
+            throw new IllegalStateException(" Quark:" + parentId + message); //$NON-NLS-1$
+        }
+        return stackDepth;
+
```

Comment:
This empty space should be removed as well as other formatting issue in this file.